### ISSUE TITLE
enhance PowerCurve to multidimensonal simulation 

### ISSUE THREAD

**Issue by palday** [palday/simr#5]
`powerCurve` currently only supports power analysis on a single variable, while varying the sample size along a single dimension (`along` or `within`).  Possible enhancements are 
- [ ] Allow `within` to accept a vector of variable names. `breaks` should then accept either a single vector argument or a list of vectors corresponding to the breaks for each variable name in `within`
- [ ] Allow `along` and `within` to be specified simultaneously. 
- [ ] Extend the plotting functions to specify whether which variable (from either `along` or `within`) option is plotted on the x-axis and which on the z-axis (i.e. distinguished by line color/type). All remaining variables specified will be used to generate facets/panels.
- [ ] Extend the `within` syntax to accept a vector of variable names.
- [ ] Add a new function `powerCurveMultiple` multiple variables to be tested simultaneously via `powerSimMultiple`. Curves can either be overplotted (i.e. variable name on the z-axis) (default) or plotted as multiple facets/panels. The awkward calling convention for the `test` parameter makes a separate function worthwhile.

**Comment by pitakakariki** [comment]
I've been thinking of writing a generic `powerPly` function (and then maybe a few wrappers for specific use cases).

In particular I'd like to make it easy to calculate power for a range of effect sizes - probably via a wrapper which passes a vector of models to `powerPly`.

I'm a little worried that this could make it easy to accidentally specify something that needs a zillion simulations because there are so many dimensions. On the other hand, performing multiple tests on a single fit could save a bit of time if that's something people want to do.

Eventually I'd like to implement something like [Hooper, Versatile sample-size calculation using simulation](http://www.stata-journal.com/article.html?article=st0282) to help people optimise over several design variables.

**Comment by palday** [comment]
I've already implemented multiple tests on a single fit as `powerSimMultiple` (including the matching `print`, `summary` and `confint` functions) in the `simultaneous` branch on my fork. It's not quite the fully general functionality of `powerPly`, but it shows how to do some of that.

It saves a bit of time, but it doesn't yet full advantage of the simultaneity built into some tests, e.g. the ones that use `car::Anova` because it reruns the test function for each variable tested. I didn't want to create a special case just for the omnibus ANOVA case but it does seem like a major loss of efficiency. All in all, it's a very naive extension, but it's still more efficient than requiring new simulations for each additional variable.

One advantage to a proper `powerPly` (that we could possibly implement even for existing internal functions) is that you could specify a parallel backend, as `plyr` allows you to do. That may make things even more memory intensive, but if you have the memory and a few cores, it may be worth your while. `plyr` does it in the `*ply` function calls, but we could do it as a generic package option for internal use.

**Comment by pitakakariki** [comment]
Parallel is fairly high on my to-do list. 90% of the work is done - I reordered the simulations, fits, and tests so that they can easily be done in parallel. The main `llply` call (in `maybe_llply`) just needs the right `.parallel` and `.paropts` arguments.

I think that means no progress bars though, which would be a shame.

**Comment by pitakakariki** [comment]
I've made a `powerPly` branch for when you're ready to push your `powerSimMultiple` branch. I'll put the new tests onto CRAN with version 1.0.2, but I'd rather wait until 1.1 for any big new user-visible functions.

**Comment by palday** [comment]
Yeah, parallel means no progress bars -- this has been a "problem" in other packages as well. 

Waiting a larger version for completely new functions sounds good. I have one more set of tests to push ...
